### PR TITLE
random spawners now check if the destination turf is unblocked in a straight line

### LINKED ISSUE #82312: Flakey Tests With Cockroaches Spawning On Icebox
```txt
[03:22:30] Runtime in code/datums/elements/body_temp_sensitive.dm,86: The cockroach loaded on in a loc with unsafe temperature at [55, 118, 4] (area : Icemoon Wastes): 180K. Acceptable Range: 270K - 1e+31K,
  proc name:  stack trace (/proc/_stack_trace)
  src: null
  call stack:
   stack trace("The cockroach loaded on in a l...", "code/datums/elements/body_temp...", 86)
  /datum/element/body_temp_sensi... (/datum/element/body_temp_sensitive): check safe environment(the cockroach (/mob/living/basic/cockroach))
  /datum/element/body_temp_sensi... (/datum/element/body_temp_sensitive): Attach(the cockroach (/mob/living/basic/cockroach), 270, 1e+31, 1, 1, 1)
  the cockroach (/mob/living/basic/cockroach):  AddElement(/list (/list))
  the cockroach (/mob/living/basic/cockroach): apply temperature requirements(1)
  the cockroach (/mob/living/basic/cockroach): on ssair init(Atmospherics (/datum/controller/subsystem/air))
  Atmospherics (/datum/controller/subsystem/air):  SendSignal("subsystem_post_initialize", /list (/list))
  Master (/datum/controller/master): init subsystem(Atmospherics (/datum/controller/subsystem/air))
  Master (/datum/controller/master): Initialize(10, 0, 1)
```

### PR DESCRIPTION
**Pull request by mc-oofert**
## About The Pull Request
random spawners now check if the destination turf is unblocked in a straight line

image

(spammed grime spawner, nothing spawned behind glass)

## Why It's Good For The Game

bug

## Changelog
not player facing

### CODE REVIEW

**Comment by mc-oofert** [comment]
wrong button

**Comment by mc-oofert** [comment]
ok it only works for the grime spawner something else is fucking up

**Comment by DaCoolBoss** [comment]
Good fix I've had trouble with these spawners before. It is player facing though, if it stops cans and stuff being generated in windows.

File: code/game/objects/effects/spawners/random/random.dm [hunk 1]
```diff
@@ -102,10 +102,21 @@
...
 				continue
+			if(!has_unblocked_line(turf_in_view))
+				continue
+
 			scatter_locations += turf_in_view
 
 	return scatter_locations
 
+/obj/effect/spawner/random/proc/has_unblocked_line(destination)
+	. = TRUE
```
Comment (Mothblocks) [inline code/game/objects/effects/spawners/random/random.dm:113]: Either return . or return TRUE normally at the end

File: code/game/objects/effects/spawners/random/random.dm [hunk 2]
```diff
@@ -102,10 +102,21 @@
...
+	var/turf/us = get_turf(src)
```
Comment (Mothblocks) [inline code/game/objects/effects/spawners/random/random.dm:114]: Could use a better name like our_turf, or I think you could also just avoid the variable entirely

**Comment by mc-oofert** [comment]
ok either something fucked up or that didnt fix it god i hate this bug

File: code/game/objects/effects/spawners/random/random.dm [hunk 3]
```diff
@@ -102,10 +102,20 @@
 		for(var/turf/turf_in_view in view(radius, get_turf(src)))
```
Comment (JohnFulpWillard) [inline code/game/objects/effects/spawners/random/random.dm:102]: The comment for this proc is very unhelpful please update it to describe what it says, returns, and what its arg does.
Also preferably make radius an early return

**Pull request merged**